src/odemis/driver/test/simulated_test.py

```diff
@@ -278,17 +278,64 @@ def test_stop(self):
         self.dev.stop()
 
         with self.assertRaises(CancelledError):
-            f1.result() # might not raise CancelledError, if the operation is not cancellable
+            f1.result()  # might not raise CancelledError, if the operation is not cancellable
             f2.result()
 
     def test_wrong_moveAbs(self):
         # wrong axis
         with self.assertRaises(ValueError):
-            self.dev.moveAbs({"ba":-2})
+            self.dev.moveAbs({"ba": -2})
 
         # wrong position
         with self.assertRaises(ValueError):
-            self.dev.moveAbs({"pressure":-5})
+            self.dev.moveAbs({"pressure": -5})
+
+
+class GenericComponentTest(unittest.TestCase):
+
+    def test_creation_complete(self):
+        comp = simulated.GenericComponent(name="test_component", role="test", vas={"vaRange":
+                                                                                       {"value": 0.1, "readonly": True,
+                                                                                        "unit": "", "range": [0, 1]},
+                                                                                   "vaChoices":
+                                                                                       {"value": 1,
+                                                                                        "choices": set(range(0, 10))},
+                                                                                   "vaBool": {"value": True}},
+                                          axes={"x": {"range": (-0.2, 0.2), "unit": "m"},
+                                                "gripper": {"choices": {'open': False, 'closed': True}}})
```

Comment:
The "choices" are reversed: it should be actual position (True/False) -> user friendly string.
This way, the `assertTrue()` will actually check something, as currently the position is "open" or "closed", which both are considered True by Python.